client/verta/verta/runtime.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+
+from collections import defaultdict
+import json
+import threading
+import os
+
+
+_THREAD = threading.local()
+
+def init_thread_context() -> None:
+    """Initialize our thread-local variable for storing logging context."""
+    if not hasattr(_THREAD, "context"):
+        _THREAD.context = defaultdict()
+
+
+def get_thread_context() -> defaultdict:
+    """
+    Return the current thread-local context or initialize a new one.
+    """
+    init_thread_context()
+    return _THREAD.context
+
+
+def set_thread_context(context: defaultdict) -> defaultdict:
+    """
+    Sets the thread-local context, overwriting any existing values.
+    """
+    # init_thread_context()
+    _THREAD.context = context
+    return _THREAD.context
+
+
+def log(**kwargs) -> None:
+    """
+    Log any number of key-value pairs to the current prediction context.
+    Context is stored in thread-local variables.
+
+    Parameters
+    ----------
+    kwargs : Any
+        Arguments and values to be added as key-value pairs to the current
+        logging context. Values must be serializable to JSON.
+
+    Returns
+    -------
+    None
+    """
+    local_context: defaultdict = get_thread_context().copy()
+    local_context.update(**kwargs)
+    set_thread_context(local_context)
```

Comment:
Out of curiosity, what reason did you have in mind for copy and reassigning `_THREAD.context` instead of

```suggestion
    get_thread_context().update(**kwargs)
```

? I'm unsure at a glance what copying will do for us, and it can be a bit of a costly operation if we don't need it.

Reply:
Removed copy operation.